### PR TITLE
refactor: use warning for invalid protocols in yarn.lock

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
We currently stop parsing if invalid protocol is found.
We need to show warning and continue parsing in these cases.

### CODE REVIEW

File: pkg/nodejs/yarn/parse.go [hunk 1]
```diff
@@ -216,6 +220,13 @@ func parseBlock(block []byte, lineNum int) (lib Library, deps []string, newLine
 		}
 	}
 
+	// in case an unsupported protocol is detected
+	// show warning and continue parsing
+	if err != nil {
+		log.Logger.Warnf(err.Error())
```
Comment (knqyf263) [inline pkg/nodejs/yarn/parse.go:226]:
```suggestion
		log.Logger.Warnf("Yarn protocol error: %s", err)
```

File: pkg/nodejs/yarn/parse.go [hunk 2]
```diff
@@ -205,7 +206,10 @@ func parseBlock(block []byte, lineNum int) (lib Library, deps []string, newLine
 			if patterns == nil || !validProtocol(protocol) {
 				skipBlock = true
 				if !ignoreProtocol(protocol) {
-					return Library{}, nil, -1, xerrors.Errorf("failed to parse package pattern: '%s', unknown protocol: '%s'", line, protocol)
+					// we need to calculate the last line of the block in order to correctly determine the line numbers of the next blocks
+					// store the error. we will handle it later
+					err = xerrors.Errorf("failed to parse package pattern: '%s', unknown protocol: '%s'", line, protocol)
```
Comment (knqyf263) [inline pkg/nodejs/yarn/parse.go:211]:
```suggestion
					err = xerrors.Errorf("unknown protocol: '%s', line: %s", protocol, line)
```

Comment (DmitriyLewen) [reply pkg/nodejs/yarn/parse.go]: Done

File: pkg/nodejs/yarn/parse.go (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/nodejs/yarn/parse.go]: Done

**Pull request merged**